Samples/SparsePackages/SparseCppApp/WinMain.cpp

```diff
@@ -0,0 +1,230 @@
+            if (FAILED(hr)) continue;
+
+            IMFActivate* pRendererActivate = nullptr;
+            hr = MFCreateVideoRendererActivate(hwnd, &pRendererActivate);
+            if (FAILED(hr)) continue;
+
+            hr = pOutputNode->SetObject(pRendererActivate);
+            if (FAILED(hr)) continue;
+
+            // Add nodes to the topology
+            hr = pTopology->AddNode(pSourceNode);
+            if (FAILED(hr)) continue;
+
+            hr = pTopology->AddNode(pOutputNode);
+            if (FAILED(hr)) continue;
+
+            hr = pSourceNode->ConnectOutput(0, pOutputNode, 0);
+            if (FAILED(hr)) continue;
+
+            pSourceNode->Release();
+            pOutputNode->Release();
+        }
+        pSD->Release();
+    }
+
+    hr = pSession->SetTopology(0, pTopology);
+    if (FAILED(hr)) return hr;
+
+    pTopology->Release();
+    pPD->Release();
+
+    // Start the session
+    PROPVARIANT varStart;
+    PropVariantInit(&varStart);
+    hr = pSession->Start(&GUID_NULL, &varStart);
+    PropVariantClear(&varStart);
+
+    return hr;
+}
+
+void Cleanup() 
+{
+    if (pSession) 
+    {
+        pSession->Close();
+        pSession->Release();
+    }
+    if (pSource) pSource->Release();
+    if (pActivate) pActivate->Release();
+    MFShutdown();
+}
+
+LRESULT CALLBACK WndProc(HWND hwnd, UINT msg, WPARAM wParam, LPARAM lParam) 
+{
+    switch (msg) 
+    {
+    case WM_DESTROY:
+        Cleanup();
+        PostQuitMessage(0);
+        break;
+    }
+    return DefWindowProc(hwnd, msg, wParam, lParam);
+}
+
+HRESULT RegisterSparsePackage(const std::wstring& externalLocation, const std::wstring& sparsePkgPath)  
+{ 
+    winrt::Windows::Management::Deployment::PackageManager packageManager;  
+    winrt::Windows::Management::Deployment::AddPackageOptions addOptions;  
+    addOptions.ExternalLocationUri(winrt::Windows::Foundation::Uri(winrt::hstring(externalLocation)));
+    auto result = packageManager.AddPackageByUriAsync(winrt::Windows::Foundation::Uri(winrt::hstring(sparsePkgPath)), addOptions).get();
+
+    return static_cast<HRESULT>(result.ExtendedErrorCode().value);
+}
+
+int WINAPI WinMain(HINSTANCE hInstance, HINSTANCE, LPSTR, int nCmdShow) 
+{
+
+    //TODO - update the value of externalLocation to match the output location of your VS Build binaries and the value of 
+    //sparsePkgPath to match the path to your signed Sparse Package (.msix). 
+    //Note that these values cannot be relative paths and must be complete paths
+    std::wstring externalLocation = L"";
+    std::wstring sparsePkgPath = L"";
+
+    //Attempt registration
+    if (!externalLocation.empty() && !sparsePkgPath.empty())
+    {
+        auto result = RegisterSparsePackage(externalLocation, sparsePkgPath);
+        if (result != HRESULT_FROM_WIN32(ERROR_PACKAGES_IN_USE) && FAILED(result))
+        {
+            MessageBox(nullptr, L"Failed to register app with identity. Running without identity", L"result", MB_ICONINFORMATION);
```

Comment:
Can we show a more helpful error message when the failure is because the user forgot/didn't know to trust any newly created self-signed certificate? `result` above has a specific error code/message for that condition.